sematic/ui/packages/main/src/utils.tsx

```diff
@@ -59,11 +61,50 @@ export function fetchJSON({
     );
 }
 
-export function useLogger() {
+export const debugFlagAtom = atomWithStorage<boolean | null>('debug', null);
+
+
+export function useDebugState() {
   const { search } = useLocation();
+  const [debugFlagLocalStorage, setDebugFlagLocalStorage] = useAtom(debugFlagAtom);
+
+
+  const debugFlagUrlValue = useMemo(
+    () => {
+      const strValue = (new URLSearchParams(search)).get('debug')?.toLocaleLowerCase();
+      if (strValue === 'true' || strValue === '1') {
+        return true;
+      };
+      if (strValue === 'false' || strValue === '0') {
+        return false;
+      };
+      return null;
+    }, [search]);
+  
+  // if debugFlagUrlValue is set, then set it in localStorage, so that it is memorized across page refreshes
+  useEffect(() => {
+    if (debugFlagUrlValue === null) {
+      return;
+    }
+    if (debugFlagLocalStorage !== debugFlagUrlValue) {
+      setDebugFlagLocalStorage(debugFlagUrlValue);
+    }
+  }, [debugFlagUrlValue, debugFlagLocalStorage, setDebugFlagLocalStorage]);
+
+  return useMemo(() => {
+    if (debugFlagUrlValue !== null) {
+      return debugFlagUrlValue;
+    }
 
-  const isLoggingExplicitlyTurnedOn = useMemo(
-    () => (new URLSearchParams(search)).has('debug'), [search]);
+    if (debugFlagLocalStorage !== null) {
+      return debugFlagLocalStorage;
+    }
+    return false; // default turning debug flag off
+  }, [debugFlagUrlValue, debugFlagLocalStorage]);
+}
+
+export function useLogger() {
```

Comment:
Please document the Dashboard dev logging and the debug flag in the developer docs.

Reply:
Yes, added to [docs/developers/manual.md](https://github.com/sematic-ai/sematic/pull/713/files#diff-465dda3f4d175981ff3e4ca70674d1425cb5e14a754629523bb2c236644c0beb) for now.